Realm/RLMObjectStore.mm

```diff
@@ -216,7 +216,18 @@ static bool RLMRealmCreateTables(RLMRealm *realm, RLMSchema *targetSchema, bool
         }
     }
 
-    // FIXME - remove deleted tables
+    if (updateExisting) {
+        NSMutableArray *tablesToDelete = [NSMutableArray array];
```

Comment:
Instead of keeping track of tables to delete, you should enumerate in reverse and delete them inline.